src/passes/Vacuum.cpp

```diff
@@ -435,6 +435,22 @@ struct Vacuum : public WalkerPass<ExpressionStackWalker<Vacuum>> {
     }
   }
 
+  void visitTryTable(TryTable* curr) {
+    // If try's body does not throw, the whole try_table can be replaced with
+    // the try's body.
+    if (!EffectAnalyzer(getPassOptions(), *getModule(), curr->body).throws()) {
+      replaceCurrent(curr->body);
+      return;
+    }
+
+    // TODO Should we attempt the same thing as in `visitTry` when the body
+    // only throws, and there is a catch_all? We cannot `nop` the `TryTable`,
+    // though. Instead, we should turn it into an unconditional `br` to the
+    // target of the `catch_all`. Also, for a `catch_all_ref`, we cannot do that
+    // optimization at all since we would not know how to create the correct
+    // `exnref`. Note: This optimization may make more sense in RemoveUnusedBrs.
```

Comment:
@aheejin The first check on `!throws` is on `curr->body`. The second check on `!hasUnremovableSideEffects` is on `curr`, which contains the body but also the try itself. And if the body can only throw then the entire try may have no effects.

Reply:
@kripken Thanks, that makes sense.
@sjrd Yeah I still think we can remove it, given that it will only work for `catch_all` and I don't expect it to be used much, at least not in C++.